app/src/main/java/com/example/weathertimeapp/activities/WeatherDetailsFragment.kt

```diff
@@ -0,0 +1,46 @@
+package com.example.weathertimeapp.activities
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.DialogFragment
+import com.example.weathertimeapp.R
+import com.example.weathertimeapp.mvp.contracts.WeatherDetailsContract
+import com.example.weathertimeapp.mvp.model.WeatherDetailsModel
+import com.example.weathertimeapp.mvp.presenter.WeatherDetailsPresenter
+import com.example.weathertimeapp.mvp.view.WeatherDetailsView
+import kotlinx.android.synthetic.main.fragment_extended_weather.view.btn_close
+
+class WeatherDetailsFragment : DialogFragment() {
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        val rootView: View = inflater.inflate(R.layout.fragment_extended_weather, container, false)
+        val cityId = this.arguments?.getInt(CITY_ID)
+        val date = this.arguments?.getString(DATE)
+        val weatherDetailPresenter: WeatherDetailsContract.Presenter =
+            WeatherDetailsPresenter(
+                WeatherDetailsModel(),
+                WeatherDetailsView(this),
+                cityId,
+                date
```

Comment:
why you construct the Presenter with id's and a date?

Reply:
is the info i'm gonna use to tell the model which city and date i need to recover to show on the dialog fragment

Reply:
Ok but it a good idea construct with this fields? you should create the Model with that info